src/backend/InvenTree/plugin/samples/integration/user_interface_sample.py

```diff
@@ -197,6 +197,18 @@ def get_ui_template_previews(self, request, context, **kwargs):
             }
         ]
 
+    def get_ui_navigation_items(self, request, context, **kwargs):
+        """Return a list of custom navigation items."""
+        return [
+            {
+                'key': 'sample-nav-item',
+                'title': 'Sample Nav Item',
+                'icon': 'ti:menu',
+                'source': self.plugin_static_file('sample_navigation.js'),
```

Comment:
Where is this source used?

Reply:
Nowhere, I got errors previously but have removed it now

Reply:
removing the source leads to me having to change the serializer used for all features - is this something that we want?

Reply:
Mh, thats interesting. But I see. In my initial implementation I have added all of these things where I didn't seen a generic usecase to an options dictionary, but this somehow was changed. Maybe we can improve this for also getting better typing by also adding a typed dict for the options. And regarding the source, best would be to make it mandatory for certain UI features and none for others that dont require a source now...